Applications/Spire/Include/Spire/Ui/StaticDropDownMenu.hpp

```diff
@@ -62,6 +62,19 @@ namespace Spire {
       //! Returns the currently selected item.
       virtual QVariant get_current_item() const;
 
+      //! Sets the current item iff the given item is currently in the menu.
+      /*!
+        \param item The current item.
+      */
+      virtual void set_current_item(const QVariant& item);
+
+      //! Sets if the list should be shown when the menu is shown.
+      /*!
+        \param is_list_shown_with_menu True iff the list should be shown when
+          the menu is shown.
+      */
+      virtual void set_list_shown_with_menu(bool is_list_shown_with_menu);
```

Comment:
Question about this class, why are so many of these methods virtual? What subclasses inherit from this? I suspect that this class needs to be reimplemented to remove all these virtual methods. As a general matter virtual methods are rarely ever used.

Reply:
The ToolbarMenu and the TrendLineStyleMenu currently inherit from it. The virtual methods could be removed based on the standard that we discussed a few days ago.

Reply:
Go ahead and remove as many virtual methods as possible. You are welcome to write interfaces where a method is pure virtual but rarely ever will a virtual method get accepted as part of an API.